src/main/scala/lambdanet/JavaAPI.scala

```diff
@@ -27,4 +34,64 @@ object JavaAPI {
   def main(args: Array[String]): Unit = {
     println("This is a test main function.")
   }
+
+  def tuplet[X, Y, Z](x: X, y: Y, z: Z): (X, Y, Z) =
+    (x, y, z)
+
+  def pair[X, Y](x: X, y: Y): (X, Y) =
+    (x, y)
+
+  def srcSpan(startLine: Int, startIndex: Int, endLine: Int, endIndex: Int, sourcePath: RelPath): SrcSpan =
+    SrcSpan((startLine, startIndex),(endLine, endIndex), sourcePath)
+
+  def userAnnotation[T](ty: T, inferred: Boolean): Annot[T] =
+    User[T](ty, inferred)
+
+  def fixed[T](ty: T): Annot[T] =
+    Fixed[T](ty)
+
+  def missing: Annot[GType] = Missing
+
+  def option[T](v: T):Option[T] = v match {
+    case Null => None
+    case _    => Some(v)
+  }
+
+  def argsCompatibility(args: Vector[(Symbol, Object, Object)]): Vector[(Symbol, TyAnnot, SrcSpan)] =
+    args.asInstanceOf[Vector[(Symbol, TyAnnot, SrcSpan)]]
+
+  def annotationCompatibility(value: Object): Annot[GType] = value.asInstanceOf[Annot[GType]]
+
+  def optionSrcSpanCompatibility(value: Object): Option[SrcSpan] = value.asInstanceOf[Option[SrcSpan]]
+
+  def predictWithGModule(model: Model,
+                         sourcePath: Path,
+                         gModules: Vector[GModule],
+                         numOfThreads: Int,
+                         predictTopK: Int,
+                        ): PredictionResults = {
+    val predictionService = model.PredictionService(numOfThreads, predictTopK)
+    predictionService.predictOnProjectWithGModules(sourcePath, gModules, warnOnErrors = false)
+  }
+
+  def typeForSrcSpanFromMap(map: Map[PNode, TopNDistribution[PType]], srcSpan: Option[SrcSpan]): String = {
```

Comment:
This code can be greatly simplified and/or optimized. We should discuss how to best do this.